test/torch/tensors/test_large_precision.py

```diff
@@ -218,3 +219,22 @@ def test_mod():
     lpt2 = x2.fix_prec(internal_type=internal_type, precision_fractional=precision_fractional)
     result = lpt1 % lpt2
     assert torch.all(torch.eq(expected, result.float_precision()))
+
+
+@pytest.mark.parametrize(
+    "x, expected",
+    [
+        (torch.tensor([1]), torch.tensor([1.0])),
+        (torch.tensor([1.0]), torch.tensor([1.0])),
+        (torch.tensor([2000.0, 1.0]), torch.tensor([2000.0, 1.0])),
+        (torch.tensor([2000.0, 1]), torch.tensor([2000.0, 1.0])),
+        (torch.tensor([-2000.0]), torch.tensor([-2000.0])),
+        (torch.tensor([-2000.0, -50]), torch.tensor([-2000.0, -50.0])),
+        (torch.tensor([-2000.0123458910]), torch.tensor([-2000.0123458910])),
```

Comment:
Could you add a test case where there are both positive and negative values in a tensor and one with a positive tensor with values after the decimal place

Reply:
Done